bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/eGateHandler.java

```diff
@@ -0,0 +1,344 @@
+                    } catch (IOException e1) {
+                        logger.warn("EGate Socket not closed {}", e1.toString());
+                    }
+                    egateSocket = null;
+                } catch (SocketException e) {
+                    logger.warn("{}", e.getLocalizedMessage());
+                    try {
+                        egateSocket.close();
+                        updateStatus(ThingStatus.OFFLINE);
+                    } catch (IOException e1) {
+                        logger.warn("EGate Socket not closed {}", e1.toString());
+                    }
+                    egateSocket = null;
+                } catch (Exception e) {
+                    logger.error(
+                            "Error while establishing connection to Dominoswiss eGate Server with Port: {} and IP: {} Error: {}",
+                            port, host, e.toString());
+                    updateStatus(ThingStatus.OFFLINE);
+                }
+                if (egateSocket != null) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+                startAutomaticRefresh();
+                logger.debug("EGate Handler connected and online, Status {} ", this.getThing().getStatus().toString());
+
+            }
+        } else {
+            logger.warn("Invalid IP address for dominoswiss eGate or wrong port : '{}'/'{}'", host, port);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Cannot connect to dominoswiss eGate gateway. host IP address or port are not set.");
+        }
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void dispose() {
+        try {
+            egateSocket.close();
+            refreshJob.cancel(true);
+            logger.debug("EGate Handler connection closed, disposing");
+        } catch (IOException e) {
+            logger.error("EGate Handler Error on dispose: {} ", e.toString());
+        }
+    }
+
+    @SuppressWarnings("null")
+    public synchronized boolean isConnected() {
+        if (egateSocket == null) {
+            return false;
+        }
+
+        // NOTE: isConnected() returns true once a connection is made and will
+        // always return true even after the socket is closed
+        // http://stackoverflow.com/questions/10163358/
+        return egateSocket.isConnected() && !egateSocket.isClosed();
+    }
+
+    /**
+     * Possible Instructions are:
+     * FssTransmit 1 Kommandoabsetzung (Controller > eGate > Dominoswiss)
+     * FssReceive 2 Empfangenes Funkpaket (Dominoswiss > eGate > Controller)
+     *
+     */
+
+    public void tiltUp(String id) {
+        for (int i = 0; i < 3; i++) {
+            pulseUp(id);
+            try {
+                Thread.sleep(150); // sleep to not confuse the blinds
+            } catch (InterruptedException e) {
+                logger.error("EGate tiltUp error: {}", e.toString());
```

Comment:
The `InterruptedException` is thrown when OH is shut-down. That means the current operation should be finished as fast as possible and no further operations should be done. You can achieve this by not catching the exception here but throw the InterruptedException and catch it at the highest level you can. As this is not an error, it shouldn't be logged, but only handled.